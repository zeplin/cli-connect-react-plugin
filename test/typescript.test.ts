import Plugin from "../src";
import { logger } from "./helper/logger";

const pluginContext = {
    components: [],
    logger,
    config: {
        tsDocgen: "react-docgen-typescript",
        tsConfigPath: "./test/tsconfig.test.json"
    }
};

describe("Connected Components React Plugin - TypeScript", () => {
    describe("Using react-docgen", () => {
        test("TSComponent.tsx snippet creation", async () => {
            const processor = new Plugin();

            const componentCode = await processor.process(
                {
                    path: "test/samples/typescript/TSComponent.tsx",
                    zeplinNames: []
                }
            );

            expect(componentCode).toMatchSnapshot();
        });

        test("TSComponentWithProps.tsx snippet creation", async () => {
            const processor = new Plugin();

            const componentCode = await processor.process(
                {
                    path: "test/samples/typescript/TSComponentWithProps.tsx",
                    zeplinNames: []
                }
            );

            expect(componentCode).toMatchSnapshot();
        });

        test("TSComponentWithChildren.tsx snippet creation", async () => {
            const processor = new Plugin();

            const componentCode = await processor.process(
                {
                    path: "test/samples/typescript/TSComponentWithChildren.tsx",
                    zeplinNames: []
                }
            );

            expect(componentCode).toMatchSnapshot();
        });

        test("TSComponentWithChildrenAndProps.tsx snippet creation", async () => {
            const processor = new Plugin();

            const componentCode = await processor.process(
                {
                    path: "test/samples/typescript/TSComponentWithChildrenAndProps.tsx",
                    zeplinNames: []
                }
            );

            expect(componentCode).toMatchSnapshot();
        });

        test("TSComponentWithImport.tsx snippet creation", async () => {
            const processor = new Plugin();

            const componentCode = await processor.process(
                {
                    path: "test/samples/typescript/TSComponentWithImport.tsx",
                    zeplinNames: []
                }
            );

            expect(componentCode).toMatchSnapshot();
        });
    });

    describe("Using react-docgen-typescript", () => {
        test("TSComponent.tsx snippet creation", async () => {
            const processor = new Plugin();

            await processor.init(pluginContext);

            processor.config = {
                tsDocgen: "react-docgen-typescript",
                tsConfigPath: "./test/tsconfig.test.json"
            };

            const componentCode = await processor.process(
                {
                    path: "test/samples/typescript/TSComponent.tsx",
                    zeplinNames: []
                }
            );

            expect(componentCode).toMatchSnapshot();
        });

        test("TSComponentWithProps.tsx snippet creation", async () => {
            const processor = new Plugin();

            await processor.init(pluginContext);

            const componentCode = await processor.process(
                {
                    path: "test/samples/typescript/TSComponentWithProps.tsx",
                    zeplinNames: []
                }
            );

            expect(componentCode).toMatchSnapshot();
        });

        test("TSComponentWithChildren.tsx snippet creation", async () => {
            const processor = new Plugin();

            await processor.init(pluginContext);

            const componentCode = await processor.process(
                {
                    path: "test/samples/typescript/TSComponentWithChildren.tsx",
                    zeplinNames: []
                }
            );

            expect(componentCode).toMatchSnapshot();
        });

        test("TSComponentWithChildrenAndProps.tsx snippet creation", async () => {
            const processor = new Plugin();

            processor.config = {
                tsDocgen: "react-docgen-typescript",
                tsConfigPath: "./test/tsconfig.test.json"
            };

            const componentCode = await processor.process(
                {
                    path: "test/samples/typescript/TSComponentWithChildrenAndProps.tsx",
                    zeplinNames: []
                }
            );

            expect(componentCode).toMatchSnapshot();
        });

        test("TSComponentWithImport.tsx snippet creation", async () => {
            const processor = new Plugin();

            await processor.init(pluginContext);

            const componentCode = await processor.process(
                {
                    path: "test/samples/typescript/TSComponentWithImport.tsx",
                    zeplinNames: []
                }
            );

            expect(componentCode).toMatchSnapshot();
        });

        test("MultiExportTSComponentWithProps.tsx snippet creation", async () => {
            const processor = new Plugin();

            await processor.init(pluginContext);

            const componentCode = await processor.process(
                {
                    path: "test/samples/typescript/MultiExportTSComponentWithProps.tsx",
                    zeplinNames: []
                }
            );

            expect(componentCode).toMatchSnapshot();
        });
    });
});
