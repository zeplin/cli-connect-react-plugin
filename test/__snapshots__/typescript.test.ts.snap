// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connected Components React Plugin - TypeScript Using react-docgen TSComponent.tsx snippet creation 1`] = `
Object {
  "description": "General component description.",
  "lang": "tsx",
  "snippet": "<MyComponent />",
}
`;

exports[`Connected Components React Plugin - TypeScript Using react-docgen TSComponentWithChildren.tsx snippet creation 1`] = `
Object {
  "description": "General component description.",
  "lang": "tsx",
  "snippet": "<MyComponent>
  {children}
</MyComponent>",
}
`;

exports[`Connected Components React Plugin - TypeScript Using react-docgen TSComponentWithChildrenAndProps.tsx snippet creation 1`] = `
Object {
  "description": "General component description.",
  "lang": "tsx",
  "snippet": "<MyComponent
  message={string}
  count={number}
  disabled={boolean}
  names={string[]}
  status={\\"waiting\\" | \\"success\\"}
  obj={object}
  obj2={{}}
  obj3={{
  id: string;
  title: string;
}}
  objArr={{
  id: string;
  title: string;
}[]}
  onSomething={Function}
  onClick={(event: React.MouseEvent<HTMLButtonElement>) => void}
  onChange={(id: number) => void}
  optional={OptionalType}>
  {children}
</MyComponent>",
}
`;

exports[`Connected Components React Plugin - TypeScript Using react-docgen TSComponentWithImport.tsx snippet creation 1`] = `
Object {
  "description": "General component description.",
  "lang": "tsx",
  "snippet": "<MyComponent
  message={string}>
  {children}
</MyComponent>",
}
`;

exports[`Connected Components React Plugin - TypeScript Using react-docgen TSComponentWithProps.tsx snippet creation 1`] = `
Object {
  "description": "General component description.",
  "lang": "tsx",
  "snippet": "<MyComponent
  message={string}
  count={number}
  disabled={boolean}
  names={string[]}
  status={\\"waiting\\" | \\"success\\"}
  obj={object}
  obj2={{}}
  obj3={{
  id: string;
  title: string;
}}
  objArr={{
  id: string;
  title: string;
}[]}
  onSomething={Function}
  onClick={(event: React.MouseEvent<HTMLButtonElement>) => void}
  onChange={(id: number) => void}
  optional={OptionalType} />",
}
`;

exports[`Connected Components React Plugin - TypeScript Using react-docgen-typescript TSComponent.tsx snippet creation 1`] = `
Object {
  "description": "General component description.",
  "lang": "tsx",
  "snippet": "<TSComponent />",
}
`;

exports[`Connected Components React Plugin - TypeScript Using react-docgen-typescript TSComponentWithChildren.tsx snippet creation 1`] = `
Object {
  "description": "General component description.",
  "lang": "tsx",
  "snippet": "<TSComponentWithChildren />",
}
`;

exports[`Connected Components React Plugin - TypeScript Using react-docgen-typescript TSComponentWithChildrenAndProps.tsx snippet creation 1`] = `
Object {
  "description": "General component description.",
  "lang": "tsx",
  "snippet": "<TSComponentWithChildrenAndProps
  message={string}
  count={number}
  disabled={boolean}
  names={string[]}
  status={'waiting' | 'success'}
  obj={object}
  obj2={{}}
  obj3={{ id: string; title: string; }}
  objArr={{ id: string; title: string; }[]}
  onSomething={Function}
  onClick={(event: MouseEvent<HTMLButtonElement, MouseEvent>) => void}
  hele={(event: MouseEvent<HTMLButtonElement, MouseEvent>) => void}
  onChange={(id: number) => void}
  optional={OptionalType} />",
}
`;

exports[`Connected Components React Plugin - TypeScript Using react-docgen-typescript TSComponentWithImport.tsx snippet creation 1`] = `
Object {
  "description": "General component description.",
  "lang": "tsx",
  "snippet": "<TSComponentWithImport
  other={string}
  message={string} />",
}
`;

exports[`Connected Components React Plugin - TypeScript Using react-docgen-typescript TSComponentWithProps.tsx snippet creation 1`] = `
Object {
  "description": "General component description.",
  "lang": "tsx",
  "snippet": "<TSComponentWithProps
  message={string}
  count={number}
  disabled={boolean}
  names={string[]}
  status={'waiting' | 'success'}
  obj={object}
  obj2={{}}
  obj3={{ id: string; title: string; }}
  objArr={{ id: string; title: string; }[]}
  onSomething={Function}
  onClick={(event: MouseEvent<HTMLButtonElement, MouseEvent>) => void}
  onChange={(id: number) => void}
  optional={OptionalType} />",
}
`;
